src/query_executor.c

```diff
@@ -19,20 +19,63 @@
 static void _returnClause_ExpandCollapsedNodes(GraphContext *gc, AST_Query *ast) {
     assert(gc);
 
-     /* Create a new return clause */
-    Vector *expandReturnElements = NewVector(AST_ReturnElementNode*, Vector_Size(ast->returnNode->returnElements));
+     /* Expanding the RETURN clause is a two phase operation:
+      * 1. Scan through every arithmetic expression within the original
+      * RETURN clause and add it to a temporary vector,
+      * if we bump into an asterisks marker indicating we should expand
+      * all nodes, relations and paths within the MATCH clause, add thoese
+      * to the temporary vector.
+      * 
+      * 2. Scanning though the temporary vector we expand each collapsed entity
+      * this will form the final RETURN clause. */
+
+    AST_ReturnNode *returnClause = ast->returnNode;
+    Vector *elementsToExpend = NewVector(AST_ReturnElementNode*, Vector_Size(returnClause->returnElements));
```

Comment:
Can we rename this Vector to `elementsToExpand`? Expend means "to use up", which would be an unnecessary confusion.